src/encoder/encoder.hpp

```diff
@@ -2,13 +2,15 @@
 
 #include <atomic>
 
+#include "capture/frame.hpp"
 #include "common/context.hpp"
+#include "common/metrics_context.hpp"
+#include "common/histogram.hpp"
```

Comment:
Seems we have mixed `#include` style here. Both `counter.hpp` and `histogram.hpp` are in common/ directory.